data/auxiliary/gather/ldap_query/ldap_queries_default.yaml

```diff
@@ -10,19 +10,42 @@ queries:
       - cACertificateDN
       - dNSHostname
       - certificateTemplates
-  - action: ENUM_ADCS_CERT_TEMPLATES
-    description: 'Enumerate ADCS certificate templates.'
+      - objectGUID
+      - caCertificate
+    referemces:
+      - https://aaroneg.com/post/2018-05-15-enterprise-ca/
+  - action: ENUM_ADCS_ESC1_VULN_TEMPLATES
+    description: 'Enumerate ADCS ESC1 vulnerable certificate templates.'
     base_dn_prefix: 'CN=Certificate Templates,CN=Public Key Services,CN=Services,CN=Configuration'
-    filter: '(objectClass=pkicertificatetemplate)'
+    filter: '(&(objectclass=pkicertificatetemplate)(!(mspki-enrollment-flag:1.2.840.113556.1.4.804:=2))(|(mspki-ra-signature=0)(!(mspki-ra-signature=*)))(|(pkiextendedkeyusage=1.3.6.1.4.1.311.20.2.2)(pkiextendedkeyusage=1.3.6.1.5.5.7.3.2) (pkiextendedkeyusage=1.3.6.1.5.2.3.4))(mspki-certificate-name-flag:1.2.840.113556.1.4.804:=1))'
```

Comment:
Not sure this is actually complete. We have client authentication, aka 1.3.6.1.5.5.7.3.2, and PKINT Client Authentication aka 1.3.6.1.5.2.3.4 and Smart Card Login (1.3.6.1.4.1.311.20.2.2), however this is also Any Purpose (2.5.29.37.0) and SubCA (not sure how to check for this).

Reply:
As a side note we do have the CA certificate manager approval check here to see if the CT_FLAG_PEND_ALL_REQUEST (0x2) is set on the `msPKI-Enrollment-Flag` attribute. If this is set all enrollment requests must be manually approved for that certificate template.

Reply:
This check also seems to not quite account for the template schema version specified in `msPKI-Tempate-Schema-Version`. If that is set to 1, the CA will require that the signing certificate has the Certificate Request Agent OID, aka 1.3.6.1.4.1.311.20.2.1, before issuing the certificate.

If alternatively the schema version is 2, then the template must set the number of authorized signatures setting, controlled via the LDAP `mspki-ra-signature` AD attribute, to an integer specifying the number of authorized signatures from enrollment agents the Certificate Signing Request (CSR) must have before the CA can start considering the permissions and make a decision on issuing a certificate. Additionally the Application Policy issuance restriction must be set to "Certificate Request Agent".

We seem to be checking for the version 2 of the certificates here but are not handling the case where the certificate template schema is version 1.

Reply:
Going to update this with the following query taken from page 64 (labeled as page 61) of the Certified PreOwned paper which addresses the above concerns minus my point on the schema template version.

```
(&(objectclass=pkicertificatetemplate)(!(mspki-enrollment-flag:1.2.840.113556.1.4.804:=2))(|(mspki-ra-signature=0)(!(mspki-ra-signature=*)))(|(pkiextendedkeyusage=1.3.6.1.4.1.311.20.2.2)(pkiextendedkeyusage=1.3.6.1.5.5.7.3.2)(pkiextendedkeyusage=1.3.6.1.5.2.3.4)(pkiextendedkeyusage=2.5.29.37.0)(!(pkiextendedkeyusage=*)))(mspki-certificate-name-flag:1.2.840.113556.1.4.804:=1))
```

Reply:
Removed this, will be in its own module in another PR. Marking resolved as a result.